lib/vector-buffers/src/config.rs

```diff
@@ -186,11 +244,18 @@ pub enum BufferType {
     /// This is more performant, but less durable. Data will be lost if Vector is restarted
     /// forcefully or crashes.
     #[configurable(title = "Events are buffered in memory.")]
-    #[serde(rename = "memory")]
+    #[serde(rename = "memory", serialize_with = "serialize_memory_config")]
```

Comment:
How does this serialize differently than what `serde` natively generates for you?

Reply:
Previously `max_events` was not an optional. Now it is, so if its not present in the schema natively the system will generate `max_bytes:null` which causes the custom deserializer to fail because it expects an integer if this key exists.

I don't believe this is an issue other then tests because that is the only case where a schema is serialized by Vector, correct me if i'm wrong though.